os/kernel/binary_manager/binary_manager.c

```diff
@@ -74,18 +80,25 @@ mqd_t binary_manager_get_mqfd(void)
  ****************************************************************************/
 int binary_manager(int argc, char *argv[])
 {
+#ifdef CONFIG_APP_BINARY_SEPARATION
 	int ret;
+#endif
 	int nbytes;
 #if !defined(CONFIG_DISABLE_SIGNALS)
 	sigset_t sigset;
 #endif
 	struct mq_attr attr;
 	binmgr_request_t request_msg;
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
 	/* Scan user binary files and Register them */
 	binary_manager_scan_ubin_all();
+#endif
 
-	ASSERT(binary_manager_get_kcount() > 0 && binary_manager_get_ucount() > 0);
+	ASSERT(binary_manager_get_kcount() > 0)
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	ASSERT(binary_manager_get_ucount() > 0);
```

Comment:
If we can check the user binary count first, then we also can merge the ifdef with the above.